polus-feature-extraction-plugin/src/main.py

```diff
@@ -8,6 +24,1003 @@
+       logger.info('Started extracting features for' + seg_file_names1)
+       
+       #Dynamically call the function based on the features required
+       feature_value = FEAT[each_feature](label_image,unitLength,intensity_image)
+
+       #get all features
+       if each_feature  == 'all':
+           #create dataframe for all features
+           df=pd.DataFrame(feature_value)
+           df = df.T
+           #if unit in metadata is considered
+           if embeddedpixelsize:
+               df.columns =['Area-%s'%img_emb_uint,'Centroid row','Centroid column','Convex area-%s'%img_emb_uint,'Eccentricity','Equivalent diameter-%s'%img_emb_uint,'Euler number','Hexagonality score','Hexagonality sd','Kurtosis','Major axis length-%s'%img_emb_uint,'Maxferet-%s'%img_emb_uint,'Maximum intensity','Mean intensity','Median','Minimum intensity','Minferet-%s'%img_emb_uint,'Minor axis length-%s'%img_emb_uint,'Mode','Neighbors','Orientation','Perimeter-%s'%img_emb_uint,'Polygonality score','Standard deviation','Skewness','Solidity']
+           #if unitlength metric is considered
+           elif unitLength and not embeddedpixelsize:
+               df.columns =['Area-%s^2'%unitLength,'Centroid row','Centroid column','Convex area-%s^2'%unitLength,'Eccentricity','Equivalent diameter-%s'%unitLength,'Euler number','Hexagonality score','Hexagonality sd','Kurtosis','Major axis length-%s'%unitLength,'Maxferet-%s'%unitLength,'Maximum intensity','Mean intensity','Median','Minimum intensity','Minferet-%s'%unitLength,'Minor axis length-%s'%unitLength,'Mode','Neighbors','Orientation','Perimeter-%s'%unitLength,'Polygonality score','Standard deviation','Skewness','Solidity']
+           #units in pixels
+           else:
+               df.columns =['Area-pixels','Centroid row','Centroid column','Convex area-pixels','Eccentricity','Equivalent diameter-pixels','Euler number','Hexagonality score','Hexagonality sd','Kurtosis','Major axis length-pixels','Maxferet-pixels','Maximum intensity','Mean intensity','Median','Minimum intensity','Minferet-pixels','Minor axis length-pixels','Mode','Neighbors','Orientation','Perimeter-pixels','Polygonality score','Standard deviation','Skewness','Solidity']
+       
+       else:
+           #create dataframe for features selected
+           df = pd.DataFrame({each_feature: feature_value})
+           if 'Area' or 'Convex area' or 'Equivalent diameter' or 'Major axis length' or 'Maxferet' or 'Minor axis length' or 'Minferet' or 'Perimeter' in df.columns:
+               if embeddedpixelsize:   
+                   df.rename({"area": "Area-%s"%img_emb_uint , "convex_area": "Convex area-%s"%img_emb_uint, "equivalent_diameter": "Equivalent diameter-%s"%img_emb_uint, "major_axis_length": "Major axis length-%s"%img_emb_uint, "minor_axis_length": "Minor axis length-%s"%img_emb_uint,"maxferet": "Maxferet-%s"%img_emb_uint, "minferet": "Minferet-%s"%img_emb_uint,"perimeter": "Perimeter-%s"%img_emb_uint},axis='columns',inplace =True)
```

Comment:
Try to format this code so that it's easier to read. Put each feature on it's own line.

For example:
```python
df.columns =[
    'Polygonality score',
    'Standard deviation',
    'Skewness',
    'Solidity']
```

Formatting code like this will make it easier to add/remove features or change feature names if desired.